### ISSUE TITLE
[azservicebus] nil pointer dereference in GetRecoveryKind()

### ISSUE THREAD

**Issue by arussellf5** [Azure/azure-sdk-for-go#17029]
- pkg: github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus
- version: v0.3.4
- go version: go 1.17

**Description**

In testing I'm seeing a nil pointer dereference error generated by [this](https://github.com/Azure/azure-sdk-for-go/blob/2bb27d4b7ddec29b5a58291c1f2206865846d377/sdk/messaging/azservicebus/internal/errors.go#L157) line within `errors.go`.

Even though other functions calling `GetRecoveryKind()` have checked to see if `err == nil`, they have not accounted for the fact that an `error` interface containing a nil pointer to an `amqp.Error` type is non-nil. So there is still a possibility of a nil pointer error here.

The following code may make the problem clearer:

```
        var detachErr *amqp.Error
	var err error
	err = detachErr

	// err is not nil, even though it contains a nil pointer
	if err == nil {
		return
	}

	var amqpErr *amqp.Error
	if errors.As(err, &amqpErr) {
		// trying to access amqpErr.Condition will cause a panic
		t.Log(amqpErr.Condition)
	}
```
It would be great if you could address this problem.

**Comment by ryepup** [comment]
I think I'm running into a similar "non-nil interface pointing at nil" issue, seeing a lot of these in my logs:

> link detached, reason: *Error(nil)

**Comment by ryepup** [comment]
This might be related to https://github.com/Azure/go-amqp/pull/123

**Comment by ryepup** [comment]
Thanks for the quick response @richardpark-msft !

**Comment by richardpark-msft** [comment]
No problem, thanks to both of you for reporting it!

@arussellf5, @ryepup: This fix has been released in v0.3.5 of azservicebus: [link](https://github.com/Azure/azure-sdk-for-go/releases/tag/sdk%2Fmessaging%2Fazservicebus%2Fv0.3.5)
